app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -352,6 +352,97 @@ protected function getQuoteById($quoteId)
         return $immutableQuote;
     }
 
+    /**
+     * Called after pre-auth order is confirmed as authorized on Bolt.
+     *
+     * @param string $orderIncrementId  customer facing order id
+     * @param object $payload           payload sent from Bolt
+     *
+     * @throws Mage_Core_Exception if there is a problem retrieving the bolt transaction reference from the payload
+     */
+    public function receiveOrder( $orderIncrementId, $payload ) {
```

Comment:
if user is under bad network or close the page, this method may not be called even when payment was successful. Where do we handle such case?

Reply:
Fortunately, this was thought about long ago and code was implement for this.
https://github.com/BoltApp/bolt-magento1/blob/975f73f7ffd27b8ba702653aaeb1aea841adca43/app/code/community/Bolt/Boltpay/controllers/ApiController.php#L71-L87


Unfortunately, by nature of the mPRs, that code is in the mPR that follows this and we did not catch the request to change this from being based on the event `bolt_boltpay_authorization_after` to a direct method call.  That update will be made in mPR.16

Reply:
updated.